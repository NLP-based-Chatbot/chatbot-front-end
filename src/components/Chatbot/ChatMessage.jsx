import React from 'react'
import { makeStyles, Box, Typography } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%",
    height: "fit-content",
    marginTop: theme.spacing(3),
  },
  container_bot: {
    maxWidth: "70%",
    borderRadius: "1.2rem 1.2rem 1.2rem 0",
    width: "fit-content",
    backgroundColor: theme.palette.primary.contrastText,
    padding: theme.spacing(2, 3),
    marginLeft: theme.spacing(2)
  },
  container_user: {
    maxWidth: "70%",
    borderRadius: "1.2rem 1.2rem 0 1.2rem",
    width: "fit-content",
    backgroundColor: theme.palette.secondary.main,
    padding: theme.spacing(2, 3),
    marginRight: theme.spacing(2),
    marginLeft: "auto"
  },
  font: {
    fontWeight: "600"
  }
}))

const ChatMessage = ({ sender, message }) => {
  const classes = useStyles()
  return (
    <Box className={classes.root}>
      {sender === 'bot' ?
        <Box className={classes.container_bot}>
          <Typography variant="body1" className={classes.font}>{message}</Typography>
        </Box>
        :
        <Box className={classes.container_user}>
          <Typography variant="body1" className={classes.font}>{message}</Typography>
        </Box>
      }
    </Box>
  )
}

export default ChatMessage
