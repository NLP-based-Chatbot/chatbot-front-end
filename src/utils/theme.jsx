import { createTheme } from '@material-ui/core';

const theme = createTheme({
  typography: {
    fontFamily: ["Poppins", "sans-serif"].join(","),
  },
  palette: {
    type: 'light',
    primary: {
      light: "#0292BE",
      main: "#3A637E",
      dark: "#1A2C38",
      contrastText: "#fff",
    },
    secondary: {
      light: "#D3EFF8",
      main: "#BBDB9B",
      dark: "#ABC4A1",
      contrastText: "#000",
    },
  },
});

theme.typography.h6 = {
  [theme.breakpoints.up('md')]: {
    fontSize: "1.5rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('md')]: {
    fontSize: "1.2rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('sm')]: {
    fontSize: "0.8rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('xs')]: {
    fontSize: "0.5rem",
    fontWeight: 500
  },
}

export default theme;