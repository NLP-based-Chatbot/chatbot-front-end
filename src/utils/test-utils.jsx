import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core";
import theme from "./theme";
import { ParallaxProvider } from "react-scroll-parallax";
import { BrowserRouter } from "react-router-dom";
import "@babel/polyfill";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import AuthReducer from '../store/slices/auth';

const customRender = (
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { auth: AuthReducer }, preloadedState }),
    ...renderOptions
  } = {}
) => {
  const AllTheProviders = ({ children }) => {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <ParallaxProvider>
            <ThemeProvider theme={theme}>{children}</ThemeProvider>;
          </ParallaxProvider>
        </BrowserRouter>
      </Provider>
    );
  };
  return render(ui, { wrapper: AllTheProviders, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
