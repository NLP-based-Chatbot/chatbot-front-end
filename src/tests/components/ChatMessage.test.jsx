/**
 * @jest-environment jsdom
 */

import { render } from "../../utils/test-utils";
import { screen } from "@testing-library/react";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import ChatMessage from "../../components/Chatbot/ChatMessage";

test("initial render", async () => {
  render(<ChatMessage />);
  screen.debug();
});

test("Display the message send by user", async () => {
  const sendMessage = jest.fn();
  render(<ChatMessage sendMessage={sendMessage} domain="telecom" sender="user" text="test message by user" />);
  expect(screen.getByTestId("user")).toBeInTheDocument();
  expect(screen.getByText("test message by user")).toBeInTheDocument();
  //screen.debug();
});

test("Display the message send by bot", async () => {
  const sendMessage = jest.fn();
  render(<ChatMessage  sendMessage={sendMessage} domain="telecom" sender="bot" type='text' text="test message by bot" />);
  expect(screen.getByTestId("bot")).toBeInTheDocument();
  expect(screen.getByText("test message by bot")).toBeInTheDocument();
  //screen.debug();
});


