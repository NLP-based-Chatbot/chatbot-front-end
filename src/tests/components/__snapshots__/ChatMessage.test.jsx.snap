// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="MuiBox-root MuiBox-root-23 makeStyles-root-19"
>
  <div
    className="MuiBox-root MuiBox-root-24 makeStyles-container_user-21"
    data-testid="user"
  >
    <p
      className="MuiTypography-root makeStyles-font-22 MuiTypography-body1"
    />
  </div>
</div>
`;
